be1-go/internal/popserver/handler/coin_test.go

```diff
@@ -0,0 +1,192 @@
+		"post_transaction_zero_amount.json",
+		"send transaction zero amount",
+		mockRepository))
+
+	// Tests that the channel rejects transactions with negative amounts
+
+	inputs = append(inputs, newFailTestHandleChannelCoin(t,
+		"post_transaction_negative_amount.json",
+		"send transaction negative amount"))
+
+	// Tests that the channel rejects Transaction with wrong id
+
+	inputs = append(inputs, newFailTestHandleChannelCoin(t,
+		"post_transaction_wrong_transaction_id.json",
+		"send transaction wrong id"))
+
+	// Tests that the channel rejects Transaction with bad signature
+
+	inputs = append(inputs, newFailTestHandleChannelCoin(t,
+		"post_transaction_bad_signature.json",
+		"send transaction bad signature"))
+
+	// Tests that the channel works correctly when it receives a transaction
+
+	inputs = append(inputs, newSuccessTestHandleChannelCoin(t,
+		"post_transaction_coinbase.json",
+		"send transaction coinbase",
+		mockRepository))
+
+	// Tests all cases
+
+	for _, i := range inputs {
+		t.Run(i.name, func(t *testing.T) {
+			errAnswer := handleChannelCoin(i.channelID, i.message)
+			if i.hasError {
+				require.NotNil(t, errAnswer)
+			} else {
+				require.Nil(t, errAnswer)
+
+				for _, s := range i.sockets {
+					require.NotNil(t, s.Msg)
+
+					var msg method.Broadcast
+					err := json.Unmarshal(s.Msg, &msg)
+					require.NoError(t, err)
+
+					require.Equal(t, i.message, msg.Params.Message)
+				}
+			}
+		})
+	}
+
+}
+
+func newSuccessTestHandleChannelCoin(t *testing.T, filename string, name string, mockRepository *repository.MockRepository) inputTestHandleChannelCoin {
+	laoID := messagedata.Hash(name)
+	var sender = "M5ZychEi5rwm22FjwjNuljL1qMJWD2sE7oX9fcHNMDU="
+	var channelID = "/root/" + laoID + "/coin"
+
+	file := filepath.Join(coinPath, filename)
+	buf, err := os.ReadFile(file)
+	require.NoError(t, err)
+
+	buf64 := base64.URLEncoding.EncodeToString(buf)
+
+	m := message.Message{
+		Data:              buf64,
+		Sender:            sender,
+		Signature:         "h",
+		MessageID:         messagedata.Hash(buf64, "h"),
+		WitnessSignatures: []message.WitnessSignature{},
+	}
+
+	mockRepository.On("StoreMessageAndData", channelID, m).Return(nil)
+
+	sockets := []*socket.FakeSocket{
+		{Id: laoID + "0"},
+		{Id: laoID + "1"},
+		{Id: laoID + "2"},
+		{Id: laoID + "3"},
+	}
+
+	errAnswer := state.AddChannel(channelID)
+	require.Nil(t, errAnswer)
+
+	for _, s := range sockets {
+		errAnswer := state.Subscribe(s, channelID)
+		require.Nil(t, errAnswer)
+	}
+
+	return inputTestHandleChannelCoin{
+		name:      name,
+		channelID: channelID,
+		message:   m,
+		hasError:  false,
+		sockets:   sockets,
+	}
+}
+
+func newFailTestHandleChannelCoin(t *testing.T, filename string, name string) inputTestHandleChannelCoin {
+	laoID := messagedata.Hash(name)
+	var sender = "M5ZychEi5rwm22FjwjNuljL1qMJWD2sE7oX9fcHNMDU="
+	var channelID = "/root/" + laoID + "/coin"
+
+	file := filepath.Join(coinPath, filename)
+	buf, err := os.ReadFile(file)
```

Comment:
as we discussed, you don't need to do this. You can directly embed the files in Go with the `embed` package.

Reply:
We still need to refactor this test in the same way as the other by using generator but we were kind of missing time to do it in this PR